tensorflow/core/kernels/mkl_pooling_ops_common.cc

```diff
@@ -15,86 +15,101 @@ limitations under the License.
-  context_.dst_fmt = static_cast<mkldnn::memory::format>(
-      context_.fwd_pd.get()->dst_primitive_desc().desc().data.format);
-
-  // create MKL-DNN internal memory object with dummy data
-  context_.src_mem.reset(new memory(
-      {{{fwdParams.src_dims}, MklDnnType<T>(), context_.src_fmt}, cpu_engine_},
-      DummyData));
-  context_.dst_mem.reset(
-      new memory(context_.fwd_pd.get()->dst_primitive_desc(), DummyData));
-
-  // for max pooling, need to return workspace(ws) for backward computing
-  if (fwdParams.alg_kind == pooling_max &&
+#ifdef ENABLE_MKLDNN_V1
+  // Store expected primitive format.
+  context_.dst_fmt = MEMORY_FORMAT::any;
+#else
+  context_.dst_fmt = static_cast<MEMORY_FORMAT>(
+      context_.fwd_pd.get()->PRIMITIVE_DESC_DST.desc().data.format);
+#endif
+
+  // Create MKL-DNN internal memory object with dummy data.
+  context_.src_mem.reset(new MEMORY_CONSTRUCTOR(
+      context_.fwd_pd.get()->PRIMITIVE_DESC_SRC, cpu_engine_, DummyData));
+  context_.dst_mem.reset(new MEMORY_CONSTRUCTOR(
+      context_.fwd_pd.get()->PRIMITIVE_DESC_DST, cpu_engine_, DummyData));
+
+  // For max pooling, need to return workspace(ws) for backward computing.
+  if (fwdParams.alg_kind == ALGORITHM::pooling_max &&
       fwdParams.prop_kind == prop_kind::forward_training) {
-    auto ws_pd = context_.fwd_pd.get()->workspace_primitive_desc().desc().data;
-    // store workspace's dims and format to create workspace tensor
-    context_.ws_fmt = static_cast<mkldnn::memory::format>(ws_pd.format);
+#ifdef ENABLE_MKLDNN_V1
+    context_.ws_mem.reset(
+        new MEMORY_CONSTRUCTOR(context_.fwd_pd.get()->PRIMITIVE_DESC_WORKSPACE,
+                               cpu_engine_, DummyData));
+    context_.net_args.push_back({{MKLDNN_ARG_SRC, *context_.src_mem},
+                                 {MKLDNN_ARG_DST, *context_.dst_mem},
+                                 {MKLDNN_ARG_WORKSPACE, *context_.ws_mem}});
+    context_.fwd.reset(new pooling_forward(*context_.fwd_pd));
+#else
+    auto ws_pd = context_.fwd_pd.get()->PRIMITIVE_DESC_WORKSPACE.desc().data;
+    // Store workspace's dims and format to create workspace tensor.
+    context_.ws_fmt = static_cast<MEMORY_FORMAT>(ws_pd.format);
     context_.ws_dims.assign(ws_pd.dims, ws_pd.dims + ws_pd.ndims);
     context_.ws_dt = static_cast<mkldnn::memory::data_type>(ws_pd.data_type);
     context_.ws_size =
-        context_.fwd_pd.get()->workspace_primitive_desc().get_size();
-    context_.ws_mem.reset(new memory(
-        context_.fwd_pd.get()->workspace_primitive_desc(), DummyData));
+        context_.fwd_pd.get()->PRIMITIVE_DESC_WORKSPACE.get_size();
+
+    context_.ws_mem.reset(
+        new memory(context_.fwd_pd.get()->PRIMITIVE_DESC_WORKSPACE, DummyData));
     context_.fwd.reset(new pooling_forward(*context_.fwd_pd, *context_.src_mem,
                                            *context_.dst_mem,
                                            *context_.ws_mem));
+#endif
```

Comment:
Nit: The #if is quite long. Let's properly comment its end.
```suggestion
#endif  // ENABLE_MKLDNN_V1
```